File: Data Structures & Algorithms/find-the-difference-of-two-arrays/submission-1.cpp
class Solution {
public:
    vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
        

        sort( nums1.begin() , nums1.end() );
        sort( nums2.begin() , nums2.end() );
        vector<int> ans;
         vector<vector<int>> ans1;
        bool is = false;
       for( int i=0; i<nums1.size(); i++ ) {
        is = false;
        for( int j=0; j<nums2.size(); j++ ) {
            if( nums1[i] == nums2[j] ) {
                is = true;
            }
        }
        if( is == false ) {
            if( ans.empty() || ans.back() != nums1[i] ) {
            ans.push_back(nums1[i]);
        }
        }
       }
       
       ans1.push_back(ans);
       ans.clear();

       for( int i=0; i<nums2.size(); i++ ) {
        is = false;
        for( int j=0; j<nums1.size(); j++ ) {
            if( nums2[i] == nums1[j] ) {
                is = true;
            }
        }
        if( is == false ) {
            if( ans.empty() || ans.back() != nums2[i] ) {
            ans.push_back(nums2[i]);
        }
        }
       }
        
        ans1.push_back(ans);

        return ans1;
       

    }
};

File: Data Structures & Algorithms/next-greater-element-i/submission-1.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        
        int[] ans = new int[nums1.length];
        int  k = 0;
        boolean is = false;
        for( int i=0; i<nums1.length; i++ ) {
             for( int j=0; j<nums2.length; j++ ) {
                if( nums1[i] == nums2[j] ) {
                    is = false;
                    while( j < nums2.length-1 ) {
                        if( nums1[i] < nums2[j+1] ) {
                            is = true;
                            ans[k++] = nums2[j+1];
                        }
                        if( is == true ) {
                            break;
                        }
                        j++;
                    }

                    if( is == false  ) {
                        ans[k++] = -1;
                    }
                }
             }
        }

        return ans;
    }
}